packages/swingset-liveslots/tools/fakeVirtualSupport.js

```diff
@@ -39,6 +44,92 @@ class FakeWeakRef {
   }
 }
 
+/** @typedef {Omit<KVStore,'set' | 'delete'> & Map<string, string>} EnhancedKVStore */
+
+/**
+ * Create a Map backed by a sorted KVStore while keeping the getNextKey method
+ * specific to a KVStore making it mostly compatible with both.
+ *
+ * Iterating over the map while mutating it is "unsupported" (entries inserted
+ * that sort before the current iteration point will be skipped).
+ *
+ * The `size` property is not supported.
+ *
+ * @param {KVStore} fakeStore
+ */
+export function makeEnhancedKVStore(fakeStore) {
+  /** @type {EnhancedKVStore} */
+  const map = harden({
+    ...fakeStore,
+    set(key, value) {
+      fakeStore.set(key, value);
+      return map;
+    },
+    delete(key) {
+      const had = fakeStore.has(key);
+      fakeStore.delete(key);
+      return had;
+    },
+    clear() {
+      for (const key of map.keys()) {
+        fakeStore.delete(key);
+      }
+    },
```

Comment:
```suggestion
    clear() {
      // Collect all keys up front to avoid pointless intermediate sorting
      // inside fakeStore.
      const keys = [...map.keys()];
      for (const key of keys) {
        fakeStore.delete(key);
      }
    },
```